src/ProgressMeter.jl

```diff
@@ -163,19 +167,20 @@ mutable struct ProgressUnknown <: AbstractProgress
     triggered::Bool
     tfirst::Float64
     tlast::Float64
-    printed::Bool        # true if we have issued at least one status update
-    desc::String         # prefix to the percentage, e.g.  "Computing..."
-    color::Symbol        # default to green
-    output::IO           # output stream into which the progress is written
+    printed::Bool           # true if we have issued at least one status update
+    desc::String            # prefix to the percentage, e.g.  "Computing..."
+    color::Symbol           # default to green
+    output::IO              # output stream into which the progress is written
     numprintedvalues::Int   # num values printed below progress in last iteration
-    enabled::Bool
+    enabled::Bool           # is the output enabled
+    showspeed::Bool         # should the output include average time per iteration
```

Comment:
It's not clear to me why these existing keywords are not presented in the docstring, but I think we should try to do so; moving the comment along with the keyword as the docstring.

Reply:
Just added info to the docstrings as well as two more sections in the README. Please feel free to merge if this all looks good!


## Showing average time per iteration

You can include an average per-iteration duration in your progress meter by setting the optional keyword argument `showspeed=true` when constructing a `Progress`, `ProgressUnknown`, or `ProgressThresh`.

```julia
x,n = 1,10
p = Progress(n; showspeed=true)
for iter = 1:10
    x *= 2
    sleep(0.5)
    ProgressMeter.next!(p; showvalues = [(:iter,iter), (:x,x)])
end
```

will yield something like:

```
Progress:  XX%|███████████████████████████           |  ETA: XX:YY:ZZ (12.34  s/it)
```

instead of

```
Progress:  XX%|███████████████████████████                         |  ETA: XX:YY:ZZ
```

## Conditionally disabling a progress meter

In addition to the `showspeed` optional keyword argument, all the progress meters also support the optional `enable` keyword argument. You can use this to conditionally disable a progress bar in cases where you want less verbose output or are using another progress bar to track progress in looping over a function that itself uses a progress bar.

```julia
function my_awesome_slow_loop(n: Integer; show_progress=true)
    p = Progress(n; enable=show_progress)
    for i in 1:n
        sleep(0.1)
        next!(p)
    end
end
const SHOW_PROGRESS_BARS = parse(Bool, get(ENV, "PROGRESS_BARS", "true"))
m = 100
# let environment variable disable outer loop progress bar
p = Progress(m; enable=SHOW_PROGRESS_BARS)
for i in 1:m
    # disable inner loop progress bar since we are tracking progress in the outer loop
    my_awesome_slow_loop(i; show_progress=false)
    next!(p)
end
```